### PR TITLE
Show personal best on song select

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

**Comment by peppy** [comment]
Is there a reason the personal best can't be clicked (to view the results screen)?

**Comment by EVAST9919** [comment]
No, just forgot about it..

**Comment by peppy** [comment]
@EVAST9919 please check changes i made, mostly just code quality.

not 100% happy with the transition but that can be fixed later on.

**Comment by peppy** [comment]
- Loading circle doesn't appear sometimes
- When it does appear, it appears at a random height.

**Review by peppy — CHANGES_REQUESTED** [review]
needs work

**Comment by EVAST9919** [comment]
> * Loading circle doesn't appear sometimes
> * When it does appear, it appears at a random height.

1. Can't reproduce
2. Fixed with latest changes (not sure if it looks good though)

**Comment by peppy** [comment]
It's still flying all over the place. I'll fix it.

**Comment by peppy** [comment]
Here's a solution:

- Move the `TopScore` inside the leaderboard. You can either use `Content` in the base leaderboard to expose an area to add arbitrary footer content, or just make a `protected Container Footer`.
- Remove the bindable TopScore as it is not needed; handle it in the scores retrieval response.
- Remove all the `AutoSizeDuration`; it's not necessary.

Note that you'll probably want to create a `SongSelectLeaderboard` which has this added logic.

File: osu.Game/Online/Leaderboards/LeaderboardScore.cs [hunk 1]
```diff
@@ -79,8 +79,8 @@ private void load()
                         {
                             Anchor = Anchor.Centre,
                             Origin = Anchor.Centre,
-                            Font = OsuFont.GetFont(size: 22, italics: true),
-                            Text = RankPosition.ToString(),
+                            Font = OsuFont.GetFont(size: 20, italics: true),
+                            Text = rank <= 999 ? rank.ToString() : rank.ToMetric(decimals: rank < 100000 ? 1 : 0),
```
Comment (iiSaLMaN) [inline osu.Game/Online/Leaderboards/LeaderboardScore.cs]: is `rank <= 999` necessary?, `123.ToMetric()` => `"123"`

**Comment by EVAST9919** [comment]
Closing due to my unability implement this properly

**Comment by peppy** [comment]
I'll take this over.

**Comment by EVAST9919** [comment]
> I'll take this over.

Thanks. I wasn't sure about this at all, so didn't want to waste our time.

**Review by peppy — DISMISSED** [review]

**Review by peppy — DISMISSED** [review]

**Comment by peppy** [comment]
@EVAST9919 i'm done with the required changes if you want to take a look.

**Comment by EVAST9919** [comment]
Sure. That was a lot faster than it could've been if I tried to implement this by myself

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**